drizzlepac/imageObject.py

```diff
@@ -128,15 +128,22 @@ def close(self):
         # else:
         #     self._image.data= None # np.array(0,dtype=self.getNumpyType(self._image.header["BITPIX"]))
 
-    def clean(self):
+    def clean(self, keep_files=None):
         """ Deletes intermediate products generated for this imageObject.
         """
         clean_files = ['blotImage','crmaskImage','finalMask',
                         'staticMask','singleDrizMask','outSky',
                         'outSContext','outSWeight','outSingle',
                         'outMedian','dqmask','tmpmask',
                         'skyMatchMask']
-
+        if keep_files:
+            for kfile in keep_files:
+                if kfile in clean_files:
+                    clean_files.remove(kfile)
+                else:
+                    log.info("Did not find intermediate product type {} to keep.".format(kfile))
+                    log.info("Valid choices are: {}".format(clean_files))
+                    
```

Comment:
Another good application for sets. Also, for every "missing" (from `clean_files`) intermediate product that should be kept, there will be a log info listing all `clean_files`. Maybe it would be better to do this only ones?

Reply:
This logic was moved from here into 'astrodrizzle.py' so that the user can be made aware of any problems with the specification of `keep_files` before the code crashes after running for some indeterminate (long?) period of time.  The change also results in only 1 message being logged for the user, not one for each missing/bad file type.